feat: use resolver function to resolve module (#535)
web-infra-dev/rslib: website/docs/en/config/lib/redirect.mdx

# lib.redirect

:::info

> Redirect is the specific configuration for bundleless mode (set [lib.bundle](/config/lib/bundle) to `false`). It will not take effect in bundle mode because, all output files are packaged into a single file. As a result, import paths do not exist and do not need to be redirected.

Reviewer: ```suggestion
Redirect is the unique configuration for bundleless mode (set [lib.bundle](/config/lib/bundle) to `false`). It will not take effect in bundle mode where all output files are packaged into a single file, eliminating the need for import path redirection.
```

```ts
const defaultRedirect = {
  js: {
    path: true,
    extension: true,
  },
  style: true,
};
```

Configure the redirect for import paths in output files. In bundleless mode, there are often needs such as using aliases or automatically appending suffixes for ESM products. The `redirect` configuration is designed to address these issues.

Common scenarios that require redirect:

> - Automatically convert tsconfig.json paths to correct relative path

Reviewer: ```suggestion
- Automatically convert `paths` in `tsconfig.json` to correct relative path
```
